WikiDrips/WikiDoc.swift

```diff
@@ -29,15 +29,24 @@ public class WikiDoc
                 return nil
             }
             
-            return initials
+            return initials.uppercased()
         }
     }
     
     init? (data: [String: AnyObject]?) {
+        if let incomingTitle = data?["title"] as? String {
+            if let incomingDate = data?["pubDate"] as? Date {
+                pubDate = incomingDate
+                title = incomingTitle
+                return
+            }
+        }
         
 //      Make sure properties are set
         title = NSLocalizedString("defaultTitle", comment: "Default title for wikiDoc")
         pubDate = Date()
```

Comment:
So I question the design of this class here. A few things...
First, our model object are usually `struct` not `class`. Structs are value types which help a great deal with multithreading issues because it eliminates the possibility of two threads writing to the same thing at the same time.

Second, in general we don't populate our models with junk default data. So we have two options: 
1. Change the properties to optionals
```
let title: String?
let pubDate: Date?
```
and let the controller layer decide what to show when values are nil. If you change the properties to optionals then you don't need a failable init (i.e. `init?`)

2. Enforce that properties are not nil. (i.e. leave title, pubDate as non-optionals). In this case, if we cannot populate properties we simply return nil from init.

In our LSM app we actually use both of these methods... the models that get created from JSON have all optional properties (method 1). But the structs that feed UI directly have some non-optional types. (method 2)

What's confusing about what you have here is that you populate properties with default values but return nil. Why should I set default properties if the object as a whole is nil? it seems the intent was method 2? If so you can simply this init to:
```
init?(data: [String: AnyObject]?) {
        guard let title = data?["title"] as? String else { return nil }
        guard let date = data?["pubDate"] as? Date else { return nil }
        
        self.title = title
        self.pubDate = date
    }
```